Make logging less "verbose"

bigkahuna443 opened the pull request:
1. Changes the default tag minimum level to `Info`.
2. Changes detour log level to Debug. If the `logdetours` debug command is used, print regardless of log level.
3. Adds `setloglevel` debug command

This makes logs much easier to read. Log levels for individual tags can still be set in mod settings or from command line, but the ones for Everest that are useful for debugging crashes should all be Info or higher already. Detour logs can be printed on demand with the `logdetours` command, so we're not really losing functionality there.

I'll probably also work on a logging page in the wiki so we can define some good practices.
EDIT: new wiki section [here](https://github.com/EverestAPI/Resources/wiki/Recommended-Practices-and-Pitfalls#logging).

Resolves #335 and resolves #336.

bigkahuna443 committed:
Make logging less "verbose"

1. Changes default threshold for logging to Info.
2. Changes detour log level to Debug.

This makes logs much easier to read. Log levels for individual tags can still be set in mod settings, but the ones for Everest that are useful for debugging crashes should all be Info or higher already. Detour logs can be printed on demand with the "logdetours" command, so we're not really losing functionality there.

I'll probably also work on a logging page in the wiki so we can define some good practices.

bigkahuna443 commented:
I'm going to look at adding a mod setting to override the minimum log level for all tags -- this seems better than having to edit tags individually in the settings file.

WEGFan commented:
You can use this to enable all logs in settings without editing tags individually, since the setting checks prefix, not full match:

```yaml
LogLevels:
  '': Verbose
```

bigkahuna443 committed:
Change default minimum log level to a setting

This lets you lower the threshold for Everest + all tags with unspecified log levels, rather than forcing you to add each individual one to the LogLevels dictionary.

Also edited LogDetours to use the level of the setting if the "logdetours" command is used, since in that case we want it to always print.

coloursofnoise reviewed (approved):
Looks good to me, as discussed on Discord this is a change that should be announced to the community alongside having a good wiki page.

bigkahuna443 committed:
Remove setting, add "setloglevel" command

Having a command to set the minimum log level has more utility and is less confusing than having a configurable fallback minimum log level.

bigkahuna443 committed:
Merge branch 'dev' into bigkahuna443/change-logging-defaults
